pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdAdvUtils.java

```diff
@@ -51,12 +51,14 @@ static boolean acknowledge(MessageIdAdv msgId, boolean individual) {
             return false;
         }
         int batchIndex = msgId.getBatchIndex();
-        if (individual) {
-            ackSet.clear(batchIndex);
```

Comment:
@Shawyeok Could you add a test to reproduce the issue that you are trying to fix?

Reply:
Sure, I'll add a test for `org.apache.pulsar.client.impl.MessageIdAdvUtils#acknowledge` in multi-thread scenario.

Reply:
Unit test added, please feel free to take a look, thanks.